crates/runix/src/flake_ref/lock.rs

```diff
@@ -0,0 +1,85 @@
+use std::str::FromStr;
+
+use derive_more::{Deref, From};
+use once_cell::sync::Lazy;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use serde_with::DeserializeFromStr;
+use thiserror::Error;
+
+use super::Timestamp;
+
+static HASH_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new("[a-f0-9]{40}").unwrap());
+
+#[derive(Debug, Deserialize, Serialize, Clone, Deref, PartialEq, Eq, From)]
+#[from(forward)]
+pub struct NarHash {
+    #[serde(rename = "narHash")]
+    pub nar_hash: String,
+}
+
+#[derive(Debug, Deserialize, Serialize, Deref, PartialEq, Eq, From, Clone)]
+#[from(forward)]
+pub struct LastModified {
+    #[serde(rename = "lastModified")]
+    pub last_modified: Timestamp,
```

Comment:
Is there a reason we need to use `LastModified` + flatten in structs instead of just using `Timestamp`?

Reply:
Honestly it might be better to just use timestamp
I made LastModified, so the timestamp and attribute name are typed but I agree thats just cognitve ballast..

Reply:
done